src/transactions/MutableTransactionResult.h

```diff
@@ -71,6 +71,10 @@ class MutableTransactionResultBase : public NonMovableOrCopyable
 {
   protected:
     std::unique_ptr<TransactionResult> mTxResult;
+    std::optional<TransactionResult> mReplaySuccessfulTransactionResult{
```

Comment:
Why are there separate variables for successful and failure results? At most one of the two can be set right? Would it make more sense/be less error prone if you just had a single "replay result"?

Reply:
Yeah that makes sense. Changed to just have one variable.